### ISSUE TITLE
Add Pundit

### ISSUE THREAD

**Issue by maebeale** [rubyforgood/mutual-aid#61]
There is an existing PR for this that needs to be finished. Please reach out to a team lead if interested! 

* Need 4 levels for now: neighbor, volunteer, admin, sys_admin
  - Display variations for these levels are not yet decided
  - `user.sys_amin?` will need to replace `user.admin?` on users/form so most users aren't able to delete accounts
* Consider removing/replacing @admin_status declaration and usage (but maybe this can be a separate issue? I would love to be able to use that var (A LOT) while developing.)

**Comment by armahillo** [comment]
What is this blocked by?

I've not set up `pundit` before but I've managed its configuration -- this'll be a fun exercise. I'll configure the stuff in this issue, but see if I can skip locking things down in the display. I *think* that can be deferred?

**Comment by maebeale** [comment]
I noticed that block this morning. I don't think anything is blocking it at this point!

**Comment by armahillo** [comment]
OK cool! I'll carry it as far as I can with the info in the issue desc

**Comment by maebeale** [comment]
* Neighbor is to help us get on track for peer-to-peer
* volunteer is unclear, but essentially is someone who would drop things off to someone else who needed groceries, for ex
* personally, I think there should be a dispatcher role where they can edit listings
* admin can see donations, finances
* sys_admins can edit system_settings, categories, system_locales, contact_methods, etc

**Comment by maebeale** [comment]
Those notes are more role based than permission based. Prob we need a hybrid?

**Comment by armahillo** [comment]
If you want to do Role based permissions, `cancancan` is good for that. 

Let me spike out an attempt with Pundit and see if I run into any friction, though.

**Comment by maebeale** [comment]
Yay! Thanks! We've needed to get Pundit going for a while now. 

Fwiw, from prior conversations, we def prefer pundit over cancancan (and I have pundit experience but limited cancancan.), but open to your findings!

**Comment by armahillo** [comment]
@maebeale Does the entire site require authorization except for the public pages? Or are there some actions that don't require being signed in? (This might be apparent from looking at the controllers, I was just wondering if you knew offhand)

**Comment by maebeale** [comment]
I was trying to design for all public in public controller, but that's not where we ended up. 

There are a few endpoints in other controllers that don't require authorization

**Comment by armahillo** [comment]
OK, thanks! :+1:

**Comment by maebeale** [comment]
Contributions/ listings/ asks/ offers/ announcements/  community_resources all have publicly accessible endpoints (pretty sure).

And there will be done that are public depending on system setting exchange_type. (Once we go peer- to-peer, for ex, contributions will be public -- or at least the page will be. We'll def want to use pundit scopes at that point to filter/authorize at record level too).
